eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/DoctrineStorage.php

```diff
@@ -222,6 +252,23 @@ public function countImageReferences($uri)
         return (int) $statement->fetchColumn();
     }
 
+    public function countDistinctImages(): int
+    {
+        $selectQuery = $this->connection->createQueryBuilder();
+        $selectQuery
+            ->select(
+                'COUNT(DISTINCT ' . $this->connection->quoteIdentifier('contentobject_attribute_id') .
+                ', ' . $this->connection->quoteIdentifier('filepath') . ')'
+            )
+            ->from($this->connection->quoteIdentifier(self::IMAGE_FILE_TABLE))
+            ->distinct()
```

Comment:
Double check what kind of query this produces please. I think it's gonna be `SELECT DISTINCT COUNT(DISTINCT ...)` which does not make a sense.